App/util/scheduleNotification.ts

```diff
@@ -0,0 +1,83 @@
+import { Notifications } from 'expo';
+import * as Permissions from 'expo-permissions';
+import { NotificationPreferenceType, ApiContext } from '../stores';
+import { NOW, NEXT_SUNDAY } from './time';
+import { useContext } from 'react';
+
+
+export async function registerForPushNotificationsAsync(selectedPreference: NotificationPreferenceType): Promise<void> {
+    const { status: existingStatus } = await Permissions.getAsync(Permissions.NOTIFICATIONS);
+    let finalStatus = existingStatus
+    if (existingStatus !== 'granted') {
+        const { status } = await Permissions.askAsync(Permissions.NOTIFICATIONS);
+        finalStatus = status
+    }
+    if (finalStatus == 'granted' || existingStatus == 'granted') {
+        setNotificationTime(selectedPreference)
+    }
+
+}
+
+
+export function scheduleNotification(selectedPreference: NotificationPreferenceType) {
+    if (selectedPreference == 'None') {
+        Notifications.cancelAllScheduledNotificationsAsync();
+    } else {
+        registerForPushNotificationsAsync(selectedPreference)
+    }
+}
+
+
+export async function setNotificationTime(selectedPreference: NotificationPreferenceType) {
+
+    // Cannot use it here 
+    // Need a way to get current cigerattes count on every notification
+    const { api } = useContext(ApiContext);
```

Comment:
Even if you could use it here, it won't be correct.

Here, you're creating a notification with today's cigarette count, but showing today's cigarette count on e.g. Sunday evening.

Reply:
True, I am going to try the solution 1, will let you know whether it works or not.